### ISSUE TITLE
[BUG] `fields` is changing the outcome of the following commands that depend on selected fields

### ISSUE THREAD

**Issue by TackAdam** [opensearch-project/sql#2739]
**What is the bug?**
When running `stats count()` operation on a nested parameter the result seems to just return 0. If the field is specified before running the count, it will show the correct value. 

**How can one reproduce the bug?**
1. Go to 'OpenSearch Playground:Query Workbench https://playground.opensearch.org/app/opensearch-query-workbench#/
2. Click on PPL and enter the PPL "source = jaeger-span-* |stats count (references.refType)" and hit Run
<img width="809" alt="Screenshot 2024-06-11 at 4 34 36 PM" src="https://github.com/opensearch-project/sql/assets/105462877/e56514dd-b910-4fba-9760-0db85db56b8f">
-Output will be 0 

Enter the PPL "source = jaeger-span-* | fields references.refType|stats count (references.refType)" and hit Run
<img width="992" alt="Screenshot 2024-06-11 at 4 36 22 PM" src="https://github.com/opensearch-project/sql/assets/105462877/d71e5e88-94e6-4ca9-b37a-0a1b1f67cab0">
-Output will be 193

**What is the expected behavior?**
The `stats count()` should return a consistent value regardless of any preceeding field selection.

**What is your host/environment?**
 - Plugins

**Do you have any screenshots?**

**Do you have any additional context?**
Add any other context about the problem.

**Comment by LantaoJin** [comment]
This may be not a bug of SQL plugin. Regarding to nested field, OpenSearch DSL query will return `0` value too. Try below DSL
```
GET jaeger-span-*/_search
{
  "size": 0,
  "aggs": {
    "count_refType": {
      "value_count": {
        "field": "references.refType"
      }
    }
  }
}
```
PPL `source = jaeger-span-* | stats count (references.refType)` equals to above DSL query. But `source = jaeger-span-* | fields references.refType | stats count (references.refType)` equals searching keyword `patients.name` first then aggregate.

**Comment by LantaoJin** [comment]
Open https://github.com/opensearch-project/OpenSearch/issues/14347 to track

**Comment by dblock** [comment]
Is this fixed with the related OpenSearch issue @LantaoJin?

[Catch All Triage - Attendees [1](https://github.com/dblock/), [2](https://github.com/krisfreedain), [3](https://github.com/andrross), [4](https://github.com/gkamat), [5](https://github.com/getsaurabh02)]

**Comment by LantaoJin** [comment]
> Is this fixed with the related OpenSearch issue @LantaoJin?

No. It's not an OpenSearch Core bug. The aggregation query on a nested field should build [Nested Aggregations](https://opensearch.org/docs/latest/aggregations/bucket/nested/) builder, but in our v2 engine, the information of nested field is missing during expression analysis. We mistakenly use an normal aggregation builder to query on a nested field, rather than nested aggregation builder. I'm investigating how to store the nested field information in expression analyzing and trigger nested aggregation builder in v2.

**Comment by LantaoJin** [comment]
#2814 is stalled, I am going to raise another PR to fix this issue for PPL first.

**Comment by LantaoJin** [comment]
Resolved by #4979
